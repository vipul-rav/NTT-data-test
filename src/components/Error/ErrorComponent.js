import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ErrorComponent extends Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  componentDidCatch(error, info) {
    // Display fallback UI
    this.setState({ hasError: true });
    // eslint-disable-next-line no-console
    console.log(error);
    // eslint-disable-next-line no-console
    console.log(info);
  }
  render() {
    if (this.state.hasError) {
      return (
        <div>
          <h2>There is technical error.</h2>
        </div>
      );
    }
    return this.props.children;
  }
}

ErrorComponent.propTypes = {
  children: PropTypes.any
};

export { ErrorComponent };
