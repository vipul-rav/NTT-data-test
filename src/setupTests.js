import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

const AllTheProviders = ({ children }) => {
  return <BrowserRouter>{children}</BrowserRouter>;
};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
