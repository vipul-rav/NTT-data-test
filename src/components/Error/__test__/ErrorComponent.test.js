import React from 'react';
import { cleanup } from '@testing-library/react';
import { render } from '../../../setupTests';
import { ErrorComponent } from '../ErrorComponent';

const setup = (overridesProps) => {
  const props = {
    ...overridesProps
  };

  const ThrowError = () => {
    throw new Error('Test');
  };

  const Child = () => (props.isError ? <ThrowError /> : <div>Child</div>);

  const { container, getByText } = render(
    <ErrorComponent>
      <Child />
    </ErrorComponent>
  );

  return {
    props,
    container,
    getByText
  };
};

describe('ErrorComponent', () => {
  afterEach(cleanup);
  it('should render', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
  it('should render with Error', () => {
    const { container } = setup({ isError: true });

    expect(container).toMatchSnapshot();
  });
});
