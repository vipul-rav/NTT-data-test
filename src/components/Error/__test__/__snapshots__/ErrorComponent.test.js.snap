// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent should render 1`] = `
<div>
  <div>
    Child
  </div>
</div>
`;

exports[`ErrorComponent should render with Error 1`] = `
<div>
  <div>
    <h2>
      There is technical error.
    </h2>
  </div>
</div>
`;
